cmd/stack_lint.go

```diff
@@ -0,0 +1,135 @@
+// Copyright © 2019 IBM Corporation and others.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"github.com/spf13/cobra"
+	"io/ioutil"
+	"os"
+	"path"
+	"strconv"
+)
+
+var lintCmd = &cobra.Command{
+	Use:   "lint",
+	Short: "Lint your stack to verify that it conforms to the standard of an Appsody stack",
+	Long: `This command will validate that your stack has the structure of an Appsody stack. It will inform you of files/directories
+missing and warn you if your stack could be enhanced.
+
+This command can be run from the base directory of your stack or you can supply a path to the stack as an argument.`,
+	RunE: func(cmd *cobra.Command, args []string) error {
+
+		stackPath := os.Getenv("PWD")
+		errorCount := 0
+		warningCount := 0
+
+		if len(args) > 0 {
+			stackPath = args[0]
+		}
+
+		imagePath := stackPath + "/image"
```

Comment:
Throughout all this code, instead of using "/" you should use filepath.Join
For example: `imagePath := filepath.Join(stackPath, "image")`